source/pele/base_potential_online.h

```diff
@@ -0,0 +1,88 @@
+#ifndef PELE_POTENTIAL_ONLINE_H
+#define PELE_POTENTIAL_ONLINE_H
+
+#include "pele/base_potential.h"
+
+namespace pele {
+    
+class BasePotentialOnline : public BasePotential {
+protected:
+    const size_t m_nr_terms;
+public:
+    BasePotentialOnline(const size_t nr_terms)
+        : m_nr_terms(nr_terms)
+    {}
+    virtual ~BasePotentialOnline() {}
+    virtual double get_energy(Array<double> x)
+    {
+        double energy = 0;
+        for (size_t i = 0; i < m_nr_terms; ++i) {
+            energy += get_energy(x, i);
+        }
+        return energy;
+    }
+    virtual double get_energy_ogradient(Array<double> x, const size_t index, Array<double> ograd)
+    {
+        numerical_ogradient(x, index, ograd);
+        return get_energy(x);
+    }
+    virtual double get_energy_ogradient_ogradient2(Array<double>x, const size_t index, Array<double> ograd, Array<double> ograd2)
```

Comment:
This is gives also the second derivatives (in addition to the gradient), but only the diagonal of the Hessian. (Check e.g. the example in the xy implementation.) The second derivatives can be used to estimate the stepsize of the stochastic gradient descent:
https://github.com/kjs73/pele/blob/e6f9e1d0a43f813c91ea289d44d0661e5640b02a/source/pele/stochastic_diagonal_levenberg_marquardt.h